src/index.ts

```diff
@@ -20,72 +20,135 @@
  */
 
 // The only import we need from the Node AuthN library is the Session class.
-import { Session, InMemoryStorage } from "@inrupt/solid-client-authn-node";
+import {
+  Session,
+  InMemoryStorage,
+  ILoginInputOptions,
+} from "@inrupt/solid-client-authn-node";
+import express from "express";
 
-const argv = require("yargs/yargs")(process.argv.slice(2))
-  .describe(
-    "oidcIssuer",
-    "The identity provider at which the user should authenticate."
-  )
-  .alias("issuer", "oidcIssuer")
-  .describe("clientName", "The name of the bootstrapped app.")
-  .demandOption(["oidcIssuer"])
-  .locale("en")
-  .help().argv;
+import {
+  promptClientName,
+  promptIdp,
+  promptPort,
+  promptRegistration,
+  promptStaticClientInfo,
+} from "./prompts";
 
-import express from "express";
-const app = express();
-const PORT = 3001;
-const iriBase = `http://localhost:${PORT}`;
-const storage = new InMemoryStorage();
+type InputOptions = {
+  solidIdentityProvider?: string;
+  clientName?: string;
+  registrationType?: "static" | "dynamic";
+};
+
+type ValidatedOptions = {
+  solidIdentityProvider: string;
+  registrationType: "static" | "dynamic";
+  clientName?: string;
+};
 
-// Initialised when the server comes up and is running...
-let session: Session;
+async function main(): Promise<void> {
+  // Get CLI arguments
+  const argv = require("yargs/yargs")(process.argv.slice(2))
+    .describe(
+      "solidIdentityProvider",
+      "The identity provider at which the user should authenticate."
+    )
+    .alias("idp", "solidIdentityProvider")
+    .describe("appName", "The name of the app you are registering.")
+    .describe(
+      "registration",
+      "[static] if you want to manually register the client, [dynamic] otherwise."
+    )
+    .describe(
+      "port",
+      "The port number on which the identity provider will return the token."
```

Comment:
Huh? What is this port number (i.e. I'm assuming it's the port number that this NodeJS server will be listening on, so I'd try and explain that). Something like:
```suggestion
      "The port number this NodeJS server should listen on (i.e. the token generation flow results in the Solid Identity Provider calling us back, and this is the port number our server will listen on, and that we tell the Solid Identity Provider to use when calling us back)."
```